pkg/lvm/lvm_util.go

```diff
@@ -254,6 +256,71 @@ func ResizeLVMVolume(vol *apis.LVMVolume, resizefs bool) error {
 	return err
 }
 
+// ResizeBTRFSVolume resizes the LVM volume as well as expand btrfs filesystem
+// Step1: Expanding LVM Volume (lvextend lvmvg -n <volume_name> -L size)
+// Step2: Expanding btrfs filesystem (btrfs filesystem resize max <mount_path>)
+func ResizeBTRFSVolume(vol *apis.LVMVolume, mountPath string) error {
+
+	desiredVolSize, err := strconv.ParseUint(vol.Spec.Capacity, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	curVolSize, err := getLVSize(vol)
+	if err != nil {
+		return err
+	}
+
+	// Expand LVM volume only when desired volume size is greater
+	// than current volume size. This will handles a case where
+	// LVM expansion is succeeded but not the FS expansion in earlier
+	// requests
+	if desiredVolSize > curVolSize {
```

Comment:
this shoud be part of NodeExpandVolume itself. Before calling expand volume we shoould check if desired size for every one, not just for btrfs.

Reply:
updated the code as discussed... PTAL